addons/sale_coupon/models/sale_order.py

```diff
@@ -141,20 +141,49 @@ def _get_reward_values_discount_percentage_per_line(self, program, line):
         discount_amount = line.product_uom_qty * line.price_reduce * (program.discount_percentage / 100)
         return discount_amount
 
+    def _get_reward_values_fixed_amount(self, program):
+
+        discount_amount = - self._get_reward_values_discount_fixed_amount(program)
+
+        lines = self._get_paid_order_lines()
+        # Remove Free Lines
+        lines = lines.filtered('price_reduce')
+        lines_total_amount = sum(lines.filtered(lambda l: l.price_total > 0).mapped('price_total'))
+        reward_lines = {}
+
+        for line in lines:
+
+            taxes = line.tax_id
+
+            line_price = line.price_reduce * line.product_uom_qty
+            # Next line is equivalent to computing the % of taxes included price then getting the related sale_price
```

Comment:
Never go over 80 char / line in comments and docstring. https://github.com/odoo/odoo/pull/83508#discussion_r796442104 might be useful